presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -522,42 +522,55 @@ protected Node visitShowCreate(ShowCreate node, Void context)
 
-                List<ConstraintSpecification> constraints = new ArrayList<>();
-                for (TableConstraint<String> constraint : connectorTableMetadata.getTableConstraintsHolder().getTableConstraints()) {
-                    if (constraint instanceof PrimaryKeyConstraint) {
-                        constraints.add(new ConstraintSpecification(constraint.getName(),
-                                constraint.getColumns().stream().collect(toImmutableList()),
-                                PRIMARY_KEY,
-                                constraint.isEnabled(),
-                                constraint.isRely(),
-                                constraint.isEnforced()));
-                    }
-                    else if (constraint instanceof UniqueConstraint) {
-                        constraints.add(new ConstraintSpecification(constraint.getName(),
-                                constraint.getColumns().stream().collect(toImmutableList()),
-                                UNIQUE,
-                                constraint.isEnabled(),
-                                constraint.isRely(),
-                                constraint.isEnforced()));
-                    }
-                    else {
-                        throw new SemanticException(NOT_SUPPORTED, "Unknown constraint type");
-                    }
-                }
-                columns.addAll(constraints);
+                columns.addAll(connectorTableMetadata.getTableConstraintsHolder().getTableConstraints()
+                        .stream()
+                        .map(constraint -> {
+                            if (constraint instanceof PrimaryKeyConstraint) {
+                                return Optional.of(new ConstraintSpecification(constraint.getName(),
+                                        constraint.getColumns().stream().collect(toImmutableList()),
+                                        PRIMARY_KEY,
+                                        constraint.isEnabled(),
+                                        constraint.isRely(),
+                                        constraint.isEnforced()));
+                            }
+                            else if (constraint instanceof UniqueConstraint) {
+                                return Optional.of(new ConstraintSpecification(constraint.getName(),
+                                        constraint.getColumns().stream().collect(toImmutableList()),
+                                        UNIQUE,
+                                        constraint.isEnabled(),
+                                        constraint.isRely(),
+                                        constraint.isEnforced()));
+                            }
+                            return Optional.empty();
+                        })
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .map(e -> (ConstraintSpecification) e)
```

Comment:
I prefer this form, but up to you

```suggestion
                        .map(ConstraintSpecification.class::cast)
```